mmeval/metrics/coco_detection.py

```diff
@@ -650,27 +644,109 @@ def compute_metric(self, results: list) -> Dict[str, Union[float, list]]:
+                if self.metric_items is None:
+                    assert len(result) == 6
+                    headers = [
+                        f'AR@{self.proposal_nums[0]}',
+                        f'AR@{self.proposal_nums[1]}',
+                        f'AR@{self.proposal_nums[2]}',
+                        f'AR_s@{self.proposal_nums[2]}',
+                        f'AR_m@{self.proposal_nums[2]}',
+                        f'AR_l@{self.proposal_nums[2]}'
+                    ]
+                else:
+                    assert len(result) == len(self.metric_items)  # type: ignore # yapf: disable # noqa: E501
+                    headers = self.metric_items  # type: ignore
+            else:
+                table_title = f' {metric} Results (%)'
+                if self.metric_items is None:
+                    assert len(result) == 6
+                    headers = [
+                        f'{metric}_mAP', f'{metric}_mAP_50',
+                        f'{metric}_mAP_75', f'{metric}_mAP_s',
+                        f'{metric}_mAP_m', f'{metric}_mAP_l'
+                    ]
+                else:
+                    assert len(result) == len(self.metric_items)
+                    headers = [
+                        f'{metric}_{item}' for item in self.metric_items
+                    ]
+            table = Table(title=table_title)
+            console = Console()
+            for name in headers:
+                table.add_column(name, justify='left')
+            table.add_row(*result)
+            redirect_string = io.StringIO()
+            with contextlib.redirect_stdout(redirect_string):
+                console.print(table, end='')
+            self.logger.info('\n' + redirect_string.getvalue())
+
+            if self.classwise and metric != 'proposal':
+                self.logger.info(
+                    f'Evaluating {metric} metric of each category...')
+                classwise_table_title = f' {metric} Classwise Results (%)'
+                classwise_result = table_results[f'{metric}_classwise_result']
+
+                num_columns = min(6, len(classwise_result) * 2)
+                results_flatten = list(itertools.chain(*classwise_result))
+                headers = ['category', f'{metric}_AP'] * (num_columns // 2)
+                results_2d = itertools.zip_longest(*[
+                    results_flatten[i::num_columns] for i in range(num_columns)
+                ])
+
+                table = Table(title=classwise_table_title)
+                console = Console()
+                for name in headers:
+                    table.add_column(name, justify='left')
+                for _result in results_2d:
+                    table.add_row(*_result)
+                redirect_string = io.StringIO()
+                with contextlib.redirect_stdout(redirect_string):
+                    console.print(table, end='')
+                self.logger.info('\n' + redirect_string.getvalue())
```

Comment:
```suggestion
                with console.capture() as capture:
                    console.print(table, end='')
                self.logger.info('\n' + capture.get())
```
There is a `rich` native api for this. Should be more stable and readable.